android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ScreenOrientationModule.java

```diff
@@ -55,47 +60,212 @@ public void onCatalystInstanceDestroy() {
+  public void getOrientationLockPlatformAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+    if (activity == null) {
+      promise.reject(ANDROID_NO_ACTIVITY, ANDROID_NO_ACTIVITY_MSG);
+      return;
+    }
+
+    promise.resolve(activity.getRequestedOrientation());
+  }
+
+  // https://stackoverflow.com/questions/10380989/how-do-i-get-the-current-orientation-activityinfo-screen-orientation-of-an-a
+  // Will not work in all cases as surface rotation is not standardized across android devices, but this is best effort
+  private String getScreenOrientation(Activity activity) throws IllegalStateException {
+    WindowManager windowManager = activity.getWindowManager();
+    int rotation = windowManager.getDefaultDisplay().getRotation();
+    DisplayMetrics dm = new DisplayMetrics();
+    windowManager.getDefaultDisplay().getMetrics(dm);
+    int width = dm.widthPixels;
+    int height = dm.heightPixels;
+    String orientation;
+    // if the device's natural orientation is portrait:
+    if ((rotation == Surface.ROTATION_0
+        || rotation == Surface.ROTATION_180) && height > width ||
+        (rotation == Surface.ROTATION_90
+            || rotation == Surface.ROTATION_270) && width > height) {
+      switch (rotation) {
+        case Surface.ROTATION_0:
+          orientation = PORTRAIT_UP;
+          break;
+        case Surface.ROTATION_90:
+          orientation = LANDSCAPE_LEFT;
+          break;
+        case Surface.ROTATION_180:
+          orientation = PORTRAIT_DOWN;
+          break;
+        case Surface.ROTATION_270:
+          orientation = LANDSCAPE_RIGHT;
+          break;
+        default:
+          throw new IllegalStateException("Unknown screen orientation.");
+      }
+    }
+
+    // if the device's natural orientation is landscape or if the device
+    // is square:
+    else {
+      switch (rotation) {
+        case Surface.ROTATION_0:
+          orientation = LANDSCAPE_LEFT;
+          break;
+        case Surface.ROTATION_90:
+          orientation = PORTRAIT_UP;
+          break;
+        case Surface.ROTATION_180:
+          orientation = LANDSCAPE_RIGHT;
+          break;
+        case Surface.ROTATION_270:
+          orientation = LANDSCAPE_RIGHT;
+          break;
+        default:
+          throw new IllegalStateException("Unknown screen orientation.");
+      }
+    }
+
+    return orientation;
+  }
+
+  // TODO: is there a shared place to put these things
+  static final String ANDROID_NO_ACTIVITY = "ANDROID_NO_ACTIVITY";
+  static final String ANDROID_NO_ACTIVITY_MSG = "There is no current activity available.";
+  static final String GENERIC_ANDROID_ERROR = "GENERIC_ANDROID_ERROR";
+
+  static final String ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK = "ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK";
+  static final String ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK_MSG = "an invalid OrientationLock was passed in";
+
+  static final String UNKNOWN = "UNKNOWN";
```

Comment:
Might make sense to define a Java enum for these (and give the enum instances a `toString()` method that returns a string value)